### PR TITLE
STMP restart for Swift Mailer

### PR DESCRIPTION
**Pull request by YOzaz**
Open for comments.
N.B. Not sure if "reset()" is required on every send - see here:
https://github.com/swiftmailer/swiftmailer/issues/490#issuecomment-87634908

### CODE REVIEW

File: src/Illuminate/Mail/Mailer.php [hunk 1]
```diff
@@ -335,10 +335,37 @@ protected function parseView($view)
...
+		{
+			// Send RESET to restart the SMTP status
+			if ($transport instanceof \Swift_Transport_AbstractSmtpTransport)
+			{
+				$transport->reset();
+			}
+
+			$transport->stop();
+		}
+		catch ( \Exception $e )
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Import this, and fix the cs too.

File: src/Illuminate/Mail/Mailer.php [hunk 2]
```diff
@@ -335,10 +335,37 @@ protected function parseView($view)
...
+	 * @return void
+	 */
+	protected function resetSwiftTransport()
+	{
+		$transport = $this->getSwiftMailerTransport();
+
+		try
+		{
+			// Send RESET to restart the SMTP status
+			if ($transport instanceof \Swift_Transport_AbstractSmtpTransport)
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: You need to import this.

File: src/Illuminate/Mail/Mailer.php [hunk 3]
```diff
@@ -467,6 +498,16 @@ public function getSwiftMailer()
 	}
 
 	/**
+	 * Get the Swift Mailer Transport instance.
+	 *
+	 * @return \Swift_Transport
+	 */
+	public function getSwiftMailerTransport()
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: should not be public

**Comment by YOzaz** [comment]
@GrahamCampbell All done. Re code style - what needs to be amended?

File: src/Illuminate/Mail/Mailer.php [hunk 4]
```diff
@@ -3,9 +3,11 @@
 use Closure;
 use Swift_Mailer;
 use Swift_Message;
+use Swift_Transport_AbstractSmtpTransport;
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Please sort these by length, not alphabetically.

File: src/Illuminate/Mail/Mailer.php [hunk 5]
```diff
@@ -3,9 +3,11 @@
...
 use SuperClosure\Serializer;
 use Psr\Log\LoggerInterface;
 use InvalidArgumentException;
+use Exception;
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Please sort these by length, not alphabetically.

File: src/Illuminate/Mail/Mailer.php [hunk 6]
```diff
@@ -335,10 +337,48 @@ protected function parseView($view)
...
 	/**
+	 * Reset Swift Mailer SMTP transport adapter
+	 *
+	 * @return void
+	 */
+	protected function resetSwiftTransport()
+	{
+		$transport = $this->getSwiftMailerTransport();
+
+		if (!isset($transport))
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Please put a single space either side of the `!`.

File: src/Illuminate/Mail/Mailer.php [hunk 7]
```diff
@@ -335,10 +337,48 @@ protected function parseView($view)
...
+
+		try
+		{
+			// Send RESET to restart the SMTP status and check if it's ready for running
+			if ($transport instanceof Swift_Transport_AbstractSmtpTransport)
+			{
+				$transport->reset();
+			}
+		}
+		catch ( Exception $e )
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: No spaces around this please.

File: src/Illuminate/Mail/Mailer.php [hunk 8]
```diff
@@ -335,10 +337,48 @@ protected function parseView($view)
...
+		{
+			// In case of failure - let's try to restart it
+			try
+			{
+				$transport->stop();
+			}
+			catch ( Exception $e )
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: same

File: src/Illuminate/Mail/Mailer.php [hunk 9]
```diff
@@ -335,10 +337,48 @@ protected function parseView($view)
...
+			$transport->start();
+		}
+	}
+
+	/**
 	 * Send a Swift Message instance.
 	 *
 	 * @param  \Swift_Message  $message
-	 * @return void
+	 * @return mixed
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: ?

Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: This needs to be `void`. We're offically saying that the return value of this method cannot be relied on, so we're using `void` to indicate this.

File: src/Illuminate/Mail/Mailer.php [hunk 10]
```diff
@@ -467,6 +511,21 @@ public function getSwiftMailer()
...
+	 * @return \Swift_Transport
+	 */
+	protected function getSwiftMailerTransport()
+	{
+		if ($this->swift instanceof Swift_Mailer)
+		{
+			return $this->swift->getTransport();
+		}
+
+		return null;
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: No need for this. Remove it please.

File: src/Illuminate/Mail/Mailer.php [hunk 11]
```diff
@@ -467,6 +511,21 @@ public function getSwiftMailer()
 	}
 
 	/**
+	 * Get the Swift Mailer Transport instance.
+	 *
+	 * @return \Swift_Transport
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Should be `\Swift_Transport|null`.

File: src/Illuminate/Mail/Mailer.php (hunk 9 above)
Comment (YOzaz) [inline src/Illuminate/Mail/Mailer.php]: Swift_Mailer->send() returns int now.

**Comment by GrahamCampbell** [comment]
Please squash to one commit when you're done.

File: src/Illuminate/Mail/Mailer.php (hunk 9 above)
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Yes, we know. We don't care about that, as I said, we're offically saying that the return value of this method cannot be relied on, so we're using `void` to indicate this.

Comment (YOzaz) [inline src/Illuminate/Mail/Mailer.php]: ok.

**Pull request closed without merging**